mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -1,200 +1,116 @@
-            for subscription in subscriptions {
-                if let SubscriptionUpdateEvent::Subscribe(sub) = subscription {
-                    subacks.remove(&sub.topic_filter);
-                }
-            }
-
-            if subacks.is_empty() {
-                debug!("command handler successfully subscribed to disconnect topic");
-                return Ok(());
-            }
+    fn handle(
+        &self,
+        broker_handle: &mut BrokerHandle,
+        publication: &ReceivedPublication,
+    ) -> Result<(), HandleEventError> {
+        let array: ServiceIdentity = serde_json::from_slice(&publication.payload)
+            .map_err(HandleEventError::ParseClientId)?;
+        if let Err(e) = broker_handle.send(Message::System(SystemEvent::AuthorizationUpdate(
+            Box::new(array),
+        ))) {
+            return Err(HandleEventError::SendAuthorizedIdentitiesToBroker(e));
         }
-    }
 
-    error!("command handler failed to subscribe to disconnect topic");
-    Err(CommandHandlerError::MissingSubacks(
-        subacks.into_iter().collect::<Vec<_>>(),
-    ))
+        info!("succeeded sending authorized identity scopes to broker",);
+        Ok(())
+    }
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum CommandHandlerError {
-    #[error("failed to receive expected subacks for command topics: {0:?}")]
-    MissingSubacks(Vec<String>),
-
-    #[error("failed to subscribe command handler to command topic")]
-    SubscribeFailure(#[from] UpdateSubscriptionError),
-
-    #[error("failed to poll client when validating command handler subscriptions")]
-    PollClientFailure(#[from] mqtt3::Error),
+pub fn init_commands() -> HashMap<String, Box<dyn Command + Send>> {
+    let mut commands: HashMap<String, Box<dyn Command + Send>> = HashMap::new();
+    commands.insert(
+        DISCONNECT_TOPIC.to_string(),
+        Box::new(Disconnect {
+            topic: DISCONNECT_TOPIC.to_string(),
+        }),
+    );
+    commands.insert(
+        AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        Box::new(AuthorizedIdentities {
+            topic: AUTHORIZED_IDENTITIES_TOPIC.to_string(),
+        }),
+    );
+    commands
+}
 
-    #[error("failed to signal shutdown for command handler")]
-    ShutdownClient(#[from] mqtt3::ShutdownError),
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ServiceIdentity {
+    #[serde(rename = "Identity")]
+    identity: String,
+    #[serde(rename = "Auth_Chain")]
+    auth_chain: Option<String>,
 }
 
 #[derive(Debug, thiserror::Error)]
-enum HandleDisconnectError {
+pub enum HandleEventError {
     #[error("failed to parse client id from message payload")]
     ParseClientId(#[from] SerdeError),
 
+    #[error("failed while sending authorized identities to broker")]
+    SendAuthorizedIdentitiesToBroker(Error),
+
     #[error("failed sending broker signal to disconnect client")]
-    SignalError(#[from] Error),
+    DisconnectSignal(Error),
 }
```

Comment:
So it is better to have this HandleEventError enum for all command handler errors, or should we have a separate Error enum for every command handler?

Reply:
We need to have a common error that the error handlers return, since the Command Trait defines an error to return